crdb/tx.go

```diff
@@ -50,6 +50,17 @@ func ExecuteTx(db *sql.DB, fn func(*sql.Tx) error) (err error) {
 	if err != nil {
 		return err
 	}
+	return ExecuteInTx(tx, func() error { return fn(tx) })
+}
+
+type Tx interface {
+	Exec(query string, args ...interface{}) (sql.Result, error)
+	Commit() error
+	Rollback() error
+}
+
+// ExecuteInTx runs fn inside tx which should already have begun.
```

Comment:
I don't know anything about gorm, but I don't see how those statements could get re-executed. It must be a requirement that the transaction have not been used (except maybe for one-time-only operations like setting the isolation level) before being passed into this function.